File: app/src/main/java/in/nisb/nisbapp/TabPagerAdapter.java
package in.nisb.nisbapp;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.PagerAdapter;
import android.view.View;

/**
 * Created by mridul on 19/3/17.
 */

class TabPagerAdapter extends FragmentPagerAdapter {

    private String fragments[] = {"Home","Events","Blog","Notifications"};

    public TabPagerAdapter(FragmentManager supportFragmentManager, Context applicationContext) {
        super(supportFragmentManager);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position){
            case 0:
                return new HomeFragment();
            case 1:
                return new EventFragment();
            case 2:
                return new BlogFragment();
            case 3:
                return new NotificationFragment();
            default:
                return null;
        }
    }

    public String getPageTitle(int position){
        return "";//fragments[position];
    }

    @Override
    public int getCount() {
        return fragments.length;
    }


}

File: app/src/main/java/in/nisb/nisbapp/TabPagerAdapter.kt
package `in`.nisb.nisbapp

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.PagerAdapter
import android.view.View

/**
 * Created by mridul on 19/3/17.
 */

internal class TabPagerAdapter(supportFragmentManager: FragmentManager, applicationContext: Context) : FragmentPagerAdapter(supportFragmentManager) {

    private val fragments = arrayOf("Home", "Events", "Blog", "Notifications")

    override fun getItem(position: Int): Fragment? {
        when (position) {
            0 -> return HomeFragment()
            1 -> return EventFragment()
            2 -> return BlogFragment()
            3 -> return NotificationFragment()
            else -> return null
        }
    }

    override fun getPageTitle(position: Int): String {
        return ""//fragments[position];
    }

    override fun getCount(): Int {
        return fragments.size
    }


}
